src/ops/array/mod.rs

```diff
@@ -311,10 +311,13 @@ impl InferenceRulesOp for Shape {
             .equals(&outputs[0].rank, 1)
             .equals(&outputs[0].shape[0], &inputs[0].rank)
 
-            .given(&inputs[0].shape, move |solver, ir: Vec<usize>| {
-                let array1:Array1<i32> = Array1::from_iter(ir.into_iter().map(|u| u as i32));
-                let tensor:Tensor = Tensor::from(array1);
-                solver.equals(&outputs[0].value, valuefact!(tensor));
+            .given(&inputs[0].shape, move |solver, shape: ShapeFact| {
+                if !shape.open && shape.dims.iter().all(|d| *d != DimFact::Any) {
+                    let shape = shape.dims.iter().map(|d| if let DimFact::Only(d)= d { *d as i32 } else { 0 }).collect();
```

Comment:
well, i do need a placeholder as the dimension is going to be marshalled into a matrix of int32. I don't link using 1 very much, as it's not obvious when you see a shape that it's a magic value. I considered -1, but thought that some operators may shoke on it, but i can try it. what do you think ?

Reply:
the alternative would be to complete the missing parts of the implementation of path and actually have set_value do something when presented with individual items :)

Reply:
implementing set_value would allow us to have TensorFact.value in the form of individual IntFact (so we use the Special value) instead of an entire Tensor that requires us to use a i32.... But we would need a lot of arithmetic for it to work (including arithmetics on partial tensors of intfacts...)

Reply:
I'll switch back to one then. zero seemed to work okay actually :)